product-service/serverless.ts

```diff
@@ -75,6 +97,82 @@ const serverlessConfiguration: Serverless = {
         },
       ],
     },
+    catalogBatchProcess: {
+      handler: 'handler.catalogBatchProcess',
+      events: [
+        {
+          sqs: {
+            batchSize: 5,
+            arn: {
+              'Fn::GetAtt': [catalogItemsQueue, 'Arn'],
+            },
+          },
+        },
+      ],
+    },
+  },
+
+  resources: {
+    Resources: {
+      [catalogItemsQueue]: {
+        Type: 'AWS::SQS::Queue',
+        Properties: {
+          QueueName: catalogItemsQueue,
+        },
+      },
+      [createProductTopic]: {
+        Type: 'AWS::SNS::Topic',
+        Properties: {
+          TopicName: createProductTopic,
+        },
+      },
+      [`${createProductTopic}SuccessSub`]: {
+        Type: 'AWS::SNS::Subscription',
+        Properties: {
+          Endpoint: newProductEmail,
+          Protocol: 'email',
+          TopicArn: {
+            Ref: createProductTopic,
+          },
+          FilterPolicy: {
+            result: ['success'],
+          },
+        },
+      },
+      [`${createProductTopic}FailSub`]: {
+        Type: 'AWS::SNS::Subscription',
+        Properties: {
+          Endpoint: newProductErrorEmail,
+          Protocol: 'email',
+          TopicArn: {
+            Ref: createProductTopic,
+          },
+          FilterPolicy: {
+            result: ['fail'],
```

Comment:
here in filter policy you require result = fail, but in new-product-notification-service you publish result = error if errors > 0

Reply:
Thanks, nice catch. 
Fixed